github-runner-provisioner/runner.go

```diff
@@ -2,35 +2,25 @@ package main
 
 import (
 	"context"
-	"github.com/aws/aws-sdk-go-v2/service/ec2"
 	"github.com/datawire/infra-actions/github-runner-provisioner/internal/aws"
+	"github.com/datawire/infra-actions/github-runner-provisioner/internal/codemagic"
 )
 
+const macM1RunnerLabel = "macOS-arm64"
+const ubuntuArm64RunnerLabel = "ubuntu-arm64"
+
 var runners = map[string]func(context.Context, string, string, bool) error{
 	macM1RunnerLabel:       createMacM1Runner,
 	ubuntuArm64RunnerLabel: createUbuntuArm64Runner,
 }
 
 func createMacM1Runner(ctx context.Context, owner string, repo string, dryRun bool) error {
-	userData, err := macM1RunnerUserData(ctx, owner, repo)
+	token, err := getGitHubRunnerToken(ctx, owner, repo)
 	if err != nil {
 		return err
 	}
 
-	params := ec2.RunInstancesInput{
-		MaxCount:                          &macM1Config.instanceCount,
-		MinCount:                          &macM1Config.instanceCount,
-		DryRun:                            &dryRun,
-		ImageId:                           &macM1Config.imageId,
-		InstanceInitiatedShutdownBehavior: macM1Config.shutdownBehavior,
-		InstanceType:                      macM1Config.instanceType,
-		KeyName:                           &macM1Config.keyName,
-		Placement:                         &macM1Config.placement,
-		UserData:                          &userData,
-		TagSpecifications:                 aws.RunnerTags(owner, repo, macM1RunnerLabel),
-	}
-
-	_, err = ec2Client.Client.RunInstances(ctx, &params)
+	err = codemagic.CreateMacM1Runner(ctx, owner, repo, token, macM1RunnerLabel, dryRun, cfg.CodeMagicToken)
```

Comment:
How do we handle the transition ? I guess we want to test the runner with the Telepresence tests before removing ec2 ?

Reply:
That's a good point.  Also, if we're keeping the AWS code maybe we should exercise it once in a while to make sure it's working.

Should both runners be exposed?